packages/js-dpp/lib/dataTrigger/rewardShareDataTriggers/createMasternodeRewardSharesDataTrigger.js

```diff
@@ -0,0 +1,86 @@
+const DataTriggerConditionError = require('../../errors/consensus/state/dataContract/dataTrigger/DataTriggerConditionError');
+const DataTriggerExecutionResult = require('../DataTriggerExecutionResult');
+
+const MAX_PERCENTAGE = 10000;
+
+/**
+ * @param {DocumentCreateTransition} documentTransition
+ * @param {DataTriggerExecutionContext} context
+ * @param {Identifier} topLevelIdentityId
+ * @return {Promise<DataTriggerExecutionResult>}
+ */
+async function createMasternodeRewardSharesDataTrigger(
+  documentTransition,
+  context,
+  topLevelIdentityId,
+) {
+  const {
+    payToId,
+    percentage,
+  } = documentTransition.getData();
+
+  const ownerId = context.getOwnerId();
+
+  const result = new DataTriggerExecutionResult();
+
+  if (percentage > MAX_PERCENTAGE) {
```

Comment:
This check is possible with JSON schema. You need to get all documents owned by masternode which already exists and summarize percentage.